### ISSUE TITLE
AttributeError: module 'torch.cuda.amp' has no attribute 'autocast'

### ISSUE THREAD

**Issue by weqvr** [WongKinYiu/ScaledYOLOv4#22]
AttributeError: module 'torch.cuda.amp' has no attribute 'autocast'

**Comment by WongKinYiu** [comment]
what is your pytorch version?
it seems your pytorch version is too old.

**Comment by huangruofei** [comment]
Have you solved this problem? I have the same problem

**Comment by huangruofei** [comment]
@weqvr

**Comment by Code-keys** [comment]
1、update your torch to 1.6+
2、return to use apex : "from apex import amp" and modify the  code related.

**Comment by weqvr** [comment]
i solve this problem by installing torch=1.7.0

**Comment by laygond** [comment]
Yes, this model makes use of `torch.cuda.amp` which is available after pytorch v1.5; however, each [pytorch version is built based on a CUDA version](https://pytorch.org/get-started/previous-versions/). For example, the maximum corresponding pytorch version for cuda 10.1 is pytorch 1.7.1 and therefore within the correct parameters (cuda 10.0 has a maximum pytorch version of 1.2). Also, this model architecture makes use of a manually installed mish-activation-function. Newer versions of [Pytorch v1.9+ include built-in mish](https://pytorch.org/docs/1.9.0/search.html?q=mish&check_keywords=yes&area=default) but you will have to change WongKinYiu's code (plus upgrade your cuda to 10.2+).
